package.json

```diff
@@ -127,22 +137,24 @@
     "libp2p-delegated-peer-routing": "^0.8.0",
```

Comment:
lots of `^` in here for <1 dependencies, seems over-optimistic about how this is going to get consumed into the future

Reply:
In the future, I would like that we get confidence in the APIs to ship v1 of all the things.

Reply:
My personal preference is to get to 1.0.0 ASAP (I've even set my default version to 1.0.0) because of the semver mess below 1 is just madness for signalling and ranges. The rules are basically "YOLO" which is useless for downstream users.
The problem with <0 is that your only signal for "breaking" changes is the minor version, but we've got `^` in here so there's no way to block breaking changes slipping in when we update one of these components. For <1, it's usually best to use `~` to account for the semver-minor bumps for breaking changes. Anyone using the current version of the library, even if they pin to the specific version, is going to get a newer, broken transitive dependency (like libp2p-delegated-peer-routing) when they do their next `npm install` because we've shipped with `^`.
Alternatively, just don't treat 1.0.0 as anything special and ship it so we can use semver properly from there on.

Reply:
Yes,I totally agree on this. The team has had a lot of concerns about shipping 1.0.0, but we need to have a plan to get to this point. I think that as part of sudo work we should create a plan on where should we be to land 1.0.0 on everything

Reply:
Please ship [v1 of things](https://github.com/ipfs/js-ipfs/issues/2657)!

> Alternatively, just don't treat 1.0.0 as anything special and ship it so we can use semver properly from there on.

Yes!

Reply:
https://github.com/libp2p/js-libp2p/issues/680

we will need to do this refactor as part of the maintenance work and will affect most of our repos. So, this might be a good opportunity to ship v1 of libp2p modules